config/v3_1_experimental/types/headers.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/coreos/ignition/v2/config/shared/errors"
+)
+
+// Parse generates standard net/http headers from the data in HTTPHeaders
+func (h HTTPHeaders) Parse() (http.Header, error) {
+	headers := http.Header{}
+	for _, headerData := range h {
+		parts := strings.SplitN(string(headerData), ":", 2)
```

Comment:
Hmm, WDYT about making this a dict of string to string instead of an array of strings to parse? E.g.

```json
"httpHeaders": {
  "MyHeader": "my-value",
  "MyOtherHeader": "my-value-2"
}
```

That way, the key/value semantic is part of the spec itself. (Note it's actually allowed to send multiple headers with the same key name, but those must be semantically equivalent to sending a single header with all the values comma-separated: https://stackoverflow.com/a/24502264).

Reply:
This is not a recommended approach for json-schema, usually objects should have a predefined set of properties like 
```json
{ 
  "httpHeaders": [ 
    { 
      "headerName": "MyHeader",
      "headerValue": "my-value"
    },
    { 
      "headerName": "MyOtherHeader",
      "headerValue": "my-value-2"
    }
  ]
}
```
Random strings in keys are not desired and allowed in extreme cases with `patternProperties`.

I personally would like to keep array of strings for the type.

Reply:
Good point. Not strongly opposed, just wanting to make sure we discuss this since it'll be part of the spec from this point on.

The main advantage to having them separate is that we get sanity-checking for free as part of schema validation.

Another approach is to use tuples, e.g. `[  ["MyHeader", "my-value"], ["MyOtherHeader", "my-value-2"] ]`. Looks like that's [supported by json-schema](https://json-schema.org/understanding-json-schema/reference/array.html#tuple-validation)?

Reply:
Yes, works for me :) I'll upload a new version soon.

Reply:
Done